libkbfs/md_journal.go

```diff
@@ -355,38 +354,38 @@ func (j mdJournal) putExtraMetadata(
 
 	extraV3, ok := extra.(*ExtraMetadataV3)
 	if !ok {
-		return errors.New("Invalid extra metadata")
+		return false, false, errors.New("Invalid extra metadata")
 	}
 
 	// TODO: We lose extraV3.wkbNew and extraV3.rkbNew here. Store
 	// it as part of the mdInfo, so we don't needlessly send it
 	// while flushing.
 
-	err := checkWKBID(j.crypto, wkbID, extraV3.wkb)
+	err = checkWKBID(j.crypto, wkbID, extraV3.wkb)
 	if err != nil {
-		return err
+		return false, false, err
 	}
 
 	err = checkRKBID(j.crypto, rkbID, extraV3.rkb)
 	if err != nil {
-		return err
+		return false, false, err
 	}
 
 	// TODO: Avoid serializing if the file already exists.
 
 	err = kbfscodec.SerializeToFile(
 		j.codec, extraV3.wkb, j.writerKeyBundleV3Path(wkbID))
 	if err != nil {
-		return err
+		return false, false, err
 	}
 
 	err = kbfscodec.SerializeToFile(
 		j.codec, extraV3.rkb, j.readerKeyBundleV3Path(rkbID))
 	if err != nil {
-		return err
+		return false, false, err
 	}
 
-	return nil
+	return extraV3.wkbNew, extraV3.rkbNew, nil
```

Comment:
This will break any existing journal entries that are trying to create a new key bundle, right, since the default is false?  It's probably a super corner case that will never happen, and probably not worth fixing, but just want to make sure I understand and that you've thought about it.